src/invariant_service.rs

```diff
@@ -707,6 +708,31 @@ where
             .collect()
     }
 
+    pub fn deposit_vara(&mut self) -> TokenAmount {
```

Comment:
The deposit token is asynchronous, while the deposit VARA is synchronous. Could this create any consistency issues, for example, when depositing one token and depositing VARA in the case of opening a position, e.g., VARA/USDC?